types/src/system/auction/mod.rs

```diff
@@ -297,7 +296,65 @@ pub trait Auction:
                     delegator_public_key.clone(),
                     *delegator.bonding_purse(),
                     amount,
-                    new_validator,
+                    None,
+                )?;
+
+                let era_end_timestamp_millis = detail::get_era_end_timestamp_millis(self)?;
+                let updated_stake = delegator.decrease_stake(amount, era_end_timestamp_millis)?;
+                if updated_stake == U512::zero() {
+                    delegators.remove(&delegator_public_key);
+                };
+                updated_stake
+            }
+            None => return Err(Error::DelegatorNotFound),
+        };
+
+        self.write_bid(validator_account_hash, bid)?;
+
+        Ok(new_amount)
+    }
+
+    /// Removes specified amount of motes (or the value from the collection altogether, if the
+    /// remaining amount is 0) from the entry in delegators map for given validator and creates a
+    /// new unbonding request to the queue, which when proccessed will redelegate the
```

Comment:
Typo: "proccessed" -> "processed".  Sentences should end in periods too.

Reply:
Fixed typo + grammar